snapcraft/internal/sources.py

```diff
@@ -137,6 +138,29 @@ def __init__(self, source, source_dir, source_tag=None,
             raise IncompatibleOptionsError(
                 'can\'t specify a source-branch for a bzr source')
 
+    def _get_environment(self):
+        # A workaround for LP: #1606203
+        env = os.environ.copy()
+
+        proxies = dict(
+            http_proxy=env.get('http_proxy', ''),
+            https_proxy=env.get('https_proxy', ''),
+        )
+
+        proxies = {key: proxies[key] for key in proxies if proxies[key]}
+
+        for key in proxies:
+            if '@' in proxies[key]:
+                logger.warning(
+                    'Removing the {!r} environment variable as '
+                    'authenticated proxies and bzr will not '
+                    'work'.format(key))
+                proxies[key] = ''
```

Comment:
then env.get will return '', if '@' will be false and the env var will remain empty.
If you would like for the key to remain unset, then maybe:

```
patched_vars = {}
for var in ('http_proxy', 'https_proxy'):
    if '@' in env.get(var, ''):
        patched_vars[var] = ''
env.update(patched_vars)
```